components/productList/FilteredProductList/SortingDropdown.tsx

```diff
@@ -0,0 +1,84 @@
+import { Menu, Transition } from "@headlessui/react";
+import { CheckIcon, ChevronDownIcon } from "@heroicons/react/solid";
+import clsx from "clsx";
+import { Fragment } from "react";
+
+import { OrderDirection, ProductOrderField } from "@/saleor/api";
+
+import { UrlSorting } from "./attributes";
+
+export interface SortingOption {
+  label: string;
+  field?: ProductOrderField;
+  direction?: OrderDirection;
+  chosen: boolean;
+}
+
+export interface SortingDropdownProps {
+  optionToggle: (field?: ProductOrderField, direction?: OrderDirection) => void;
+  chosen: UrlSorting | null;
+}
+
+export function SortingDropdown({ optionToggle, chosen }: SortingDropdownProps) {
+  const options: SortingOption[] = [
+    { label: "Popularity", chosen: false },
+    { label: "Name ascending", field: "NAME", direction: "ASC", chosen: false },
+    { label: "Name descending", field: "NAME", direction: "DESC", chosen: false },
+  ];
+
+  let isChosenSet = false;
+  for (const option of options) {
+    if (option.field === chosen?.field && option.direction === chosen?.direction) {
+      option.chosen = true;
+      isChosenSet = true;
+      break;
+    }
+  }
+  if (!isChosenSet) {
+    options[0].chosen = true;
+  }
```

Comment:
The logic is fine but I'd move it outside the component and call it: `getSortingOptions(chosen)`